File: ecma6/euler007.js
/**
 * Created by gcharles on 4/15/16.
 */
/**
 *  By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.
 *
 *  What is the 10 001st prime number?
 */
var euler007 = new EulerProblem({
   problem: 7,
   testInput: 6,
   testOutput: 13,
   realInput: 10001,
   realOutput: 104743,

   solver: function (n) {
      return primes.PrimeGenerator(n).next().value;
   }
});

File: python/euler007.py
"""
By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6^(th) prime is 13.

What is the 10001^(st) prime number?
"""
from common.euler_lib import prime_generator
from euler import euler_problem


@euler_problem()
def euler007(n: int | str) -> int:
    gen = prime_generator()
    for i in range(int(n) - 1):
        next(gen)

    return next(gen)


if __name__ == '__main__':
    print(euler007(6))
    print(euler007(10_001))
